enhancements/storage/csi-vsphere-operator.md

```diff
@@ -0,0 +1,178 @@
+* For UPI install - we can additionally document required HW version while creating VMs.
+
+### Deployment strategy
+
+vSphere CSI driver operator will be deployed by [cluster-storage-operator](https://github.com/openshift/cluster-storage-operator/)
+
+#### Operator deployment via cluster-storage-operator
+
+The cluster-storage-operator will deploy all the resources necessary for creating the vSphere CSI driver operator.
+
+1. vSphere CSI driver operator will be deployed in namespace `openshift-cluster-csi-drivers`.
+2. A service account will be created for running the operator.
+3. The operator will get RBAC rules necessary for running the operator and its operand (i.e the actual vSphere CSI driver).
+4. A deployment will be created for starting the operator and will be managed by cluster-storage-operator.
+5. A instance of `ClusterCSIDriver` will be created to faciliate managment of driver operator.  `ClusterCSIDriver` is already defined in - https://github.com/openshift/api/blob/master/operator/v1/types_csi_cluster_driver.go but needs to be expanded to include vSphere CSI driver.
+6. cluster-storage-operator will request CVO to create required cloud-credentials for talking with vCenter API.
+
+#### Driver deployment via vsphere-csi-driver-operator.
+
+The operator itself will be responsible for running the driver and all the required sidecars (attacher, provisioner etc).
+
+1. The operator will assume namespace `openshift-cluster-csi-drivers` is already created for the driver.
+2. A service account will be created for running the driver.
+3. The operator will create RBAC rules necessary for running the driver and sidecars.
+4. A deployment will be created and managed by operator to handle control-plane sidecars and controller-plane driver deployment with controller-side of CSI services.
+5. A DaemonSet will be created and managed by operator to run node side of driver operations.
+6. A `CSIDriver` object will be created to expose driver features to control-plane.
+7. The driver operator will use and expose cloud-credentials created by CVO.
+
+Most of the steps outlined above is common to all CSI driver operator and vSphere CSI driver is not unique among those aspects.
+
+#### Additional consideration for vSphere
+
+There are certain aspects of driver which require special handling:
+
+##### StoragePolicy configuration
+
+Currently while deploying Openshift a user can configure datastore used by OCP via install-config.yaml. vSphere CSI driver however can't use datastore directly and must be configured with vSphere storage policy. 
+
+To solve this problem vsphere CSI operator is going to create a storagePolicy by tagging selected datastore in the installer. This will require OCP to have expanded permissions of creating storagePolicies. After creating the storagePolicy, the vSphere CSI operator will also create corresponding storageclass.
```

Comment:
> tagging selected datastore in the installer

1. Tagging datastore is not mentioned in the installer chapter above
2. Installer is not running during upgrade. To upgrade a running cluster, something must tag the datastore and I would expect it's the CSI driver operator.

Reply:
that is why we moved tagging and create storagepolicy work to operator and not in the installer.